src/lib.rs

```diff
@@ -469,23 +469,26 @@ pub fn get_configuration_file_path<'a>(
     config_name: impl Into<Option<&'a str>>,
 ) -> Result<PathBuf, ConfyError> {
     let config_name = config_name.into().unwrap_or("default-config");
-    let project = ProjectDirs::from("rs", "", app_name).ok_or_else(|| {
-        ConfyError::BadConfigDirectory("could not determine home directory path".to_string())
+    let project = choose_app_strategy(AppStrategyArgs {
+        top_level_domain: "rs".to_string(),
+        author: "".to_string(),
+        app_name: app_name.to_string(),
+    })
+    .map_err(|e| {
+        ConfyError::BadConfigDirectory(format!("could not determine home directory path: {e}"))
     })?;
 
     let config_dir_str = get_configuration_directory_str(&project)?;
 
-    let path = [config_dir_str, &format!("{config_name}.{EXTENSION}")]
+    let path = [config_dir_str, format!("{config_name}.{EXTENSION}")]
         .iter()
         .collect();
 
     Ok(path)
 }
 
-fn get_configuration_directory_str(project: &ProjectDirs) -> Result<&str, ConfyError> {
-    let path = project.config_dir();
-    path.to_str()
-        .ok_or_else(|| ConfyError::BadConfigDirectory(format!("{path:?} is not valid Unicode")))
+fn get_configuration_directory_str(project: &impl AppStrategy) -> Result<String, ConfyError> {
+    Ok(project.config_dir().display().to_string())
```

Comment:
```suggestion
fn get_configuration_directory_str(project: &impl AppStrategy) -> Result<PathBuf, ConfyError> {
    Ok(project.config_dir())
```

I know it is a private function, but maybe it is better to use the PathBuf directly instead of relying on the String approximation?
Especially since you want to do a major version.

Reply:
Also, I think the function is used only once… it was a nice helper but may be useless now?